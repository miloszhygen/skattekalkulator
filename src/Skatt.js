import React, { Component } from 'react';

//Import utils
import { calculateTax } from './utils/taxUtil'
import { urlSearchParameterUtil } from './utils/urlSearchParameterUtil'
import './Skatt.css';

// TODO: Proptypes

class Skatt extends Component {
  state = {
    taxToPay: {}
  }

  componentWillMount(){
    const responseStatus = urlSearchParameterUtil(window.location.search);
    this.setState({ taxToPay: calculateTax(responseStatus.income) })
  }

  render() {
    const {
      taxToPay: {
        income = 0,
        tax = 0,
        socialSecurityTax = 0,
        minstefradrag = 0,
        trinnTax:{
          totalTrinnSkatt = 0,
          trinnOne = 0,
          trinnTwo = 0,
          trinnThree = 0,
          trinnFour = 0
        }={}
      }={}
    } = this.state;
    return (
      <div>
        <header>
          <h1>Skattekalkulator 2018</h1>
          <h3>Skattekalkulator for allminnelig lønnsintekt*</h3>
        </header>
        <label>
          Skriv inn din bruttoinntekt 2018: <br/>
          <input
            type="text"
            name="income"
            value={income || ''}
            onChange={(e)=> this.setState({taxToPay: calculateTax(e.target.value)}) }
          />
        </label>
        <hr/>
        { (income > 0 ) &&
          <div>
            <h3>Skatt: {tax} kr</h3>
            <hr/>
            <p>Hvorav:</p>
            <p>
              <b> Trygdeavgift: {socialSecurityTax} kr</b>
            </p>
            <p>Trinnskatt: {totalTrinnSkatt} kr</p>
            {(totalTrinnSkatt > 0) &&
              <div className="box--margin-left">
                <p>Fordelt slik:</p>
                {(trinnOne > 0) && <p> Trinn 1: {trinnOne} kr</p>}
                {(trinnTwo > 0) && <p> Trinn 2: {trinnTwo} kr</p>}
                {(trinnThree > 0) && <p> Trinn 3: {trinnThree} kr</p>}
                {(trinnFour > 0) && <p> Trinn 3: {trinnFour} kr</p>}
              </div>
            }
            <p>Minstefradrag: {minstefradrag} kr</p>
          </div>
        }
        <small>*Kalkulatoren gjelder ikke bosatte i Finmark og Nord-Troms. Den tar heller ikke med i kalkulasjonene særfradrag enslige forsørgere.</small>
      </div>
    );
  }
}

export default Skatt;
