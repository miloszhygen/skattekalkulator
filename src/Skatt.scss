html{
  font-family: "Roboto";
}

.box--margin-left{
  margin-left: 20px;
}

h1 {
  font-weight:500;
  font-size:3em;
  word-break: break-all;
}