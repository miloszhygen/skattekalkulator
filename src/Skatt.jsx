import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Import component
import Helpers from './components/Helpers'
import Input from './components/Input'
import TaxCalculations from './components/TaxCalculations'

// Import utils
import { calculateTax } from './utils/taxUtil'
import { urlSearchParameterUtil } from './utils/urlSearchParameterUtil'
import { splitNumberOnKiloUtil } from './utils/splitNumberOnKiloUtil'

// Import styles
import './Skatt.scss';

// Import helpers
import { YEARLY, MONTHLY } from './helpers/setTypes'
import setEnvironment from './helpers/setEnvironment';
import {getShareUrl} from './utils/getShareUrl';


// TODO: Proptypes

class Skatt extends Component {
  constructor () {
    super()
    const {income = 0,formue = 0, finnmark, married, fradrag = 0, kapital = 0, monthly = false} = urlSearchParameterUtil(window.location.search);
    this.state = {
      stateFormue: formue,
      stateIncome: parseInt(income, 10),
      stateMarried: ( married === 'true' ),
      stateFinnmark: ( finnmark === 'true' ),
      stateFradrag: parseInt(fradrag, 10),
      stateKapital: parseInt(kapital, 10),
      taxToPay: {},
      incomePr: (monthly === 'true') ? MONTHLY : YEARLY
    }
  }

  componentDidMount () {
    this.updateSkatt()
  }
  
  componentDidUpdate (prevProps, prevState) {
    if (JSON.stringify(prevState).localeCompare(JSON.stringify(this.state))) {
      this.updateSkatt()
    }
  }

  updateSkatt () {
    const {stateIncome,stateFormue,stateFinnmark,stateMarried,stateFradrag,stateKapital,incomePr} = this.state;
    this.setState({
      shareUrl: getShareUrl(this.state),
      taxToPay: calculateTax({
        income: stateIncome,
        nettoFormue: stateFormue,
        finnmarksfradrag: stateFinnmark,
        married: stateMarried,
        fradrag: stateFradrag,
        kapital: stateKapital,
        incomePr: incomePr
      })})
  }


  render () {
    const {
      shareUrl,
      stateFinnmark,
      stateIncome,
      stateFormue,
      stateFradrag,
      stateMarried,
      stateKapital,
      taxToPay: {
        income = 0,
      }={}
    } = this.state;
    return (
      <div>
        {(setEnvironment()) && <Helpers/> }

        <header>
          <h1>Skattekalkulator 2018</h1>
          <h3>Skattekalkulator for allminnelig lønnsintekt*</h3>
        </header>


        <Input
          title="Skriv inn din bruttoinntekt 2018:"
          type="text"
          name="income"
          value={stateIncome || ''}
          onChange={(e)=> this.setState({stateIncome: parseInt(e.target.value,10) || 0})}
        />
        <Input
          titleright="Årlig"
          type="radio"
          value={YEARLY}
          checked={this.state.incomePr === YEARLY}
          onChange={(e) => this.setState({ incomePr: e.target.value })}
        />
        <Input
          titleright="Månedlig"
          type="radio"
          value={MONTHLY}
          checked={this.state.incomePr === MONTHLY}
          onChange={(e) => this.setState({ incomePr: e.target.value })}
        />

        <br/>
        {(this.state.incomePr === MONTHLY && income > 0) &&
          <div>Årlig inntekt: {splitNumberOnKiloUtil(income)} kr</div>
        }
        <hr/>

        <Input
          titleright="Rett til Finnmarksfradrags"
          type="checkbox"
          value="1000"
          checked={stateFinnmark}
          onChange={()=> this.setState({stateFinnmark:!stateFinnmark})}
        />
        <Input
          titleright="Gift"
          type="checkbox"
          value="1000"
          checked={stateMarried}
          onChange={()=> this.setState({stateMarried: !stateMarried})}
        />
        <br/>

        <Input
          title="Rente og kapitalinntekter"
          type="text"
          name="formue"
          value={stateKapital || ''}
          onChange={(e)=> this.setState({stateKapital: parseInt(e.target.value,10) || 0  })}
          text="Rente- og kapitalinntekter er typisk renten på bankinnskudd, avkastning på pengemarkedsfond og obligasjonsfond, samt gevinst ved salg av aksjer og aksjefond."
        />

        <br/>
        <Input
          title="Din netto formue"
          type="text"
          name="formue"
          value={stateFormue || ''}
          onChange={(e)=> this.setStsate({stateFormue: parseInt(e.target.value,10) || 0 })}
          text="Det er netto formue som skal legges inn. Fra og med skatteåret 2017 er det visse formuesobjekter som får en verdsettingsrabatt, samt at disse får tilordnet gjeld. Det er formuen etter fradraget for verdsettingsrabatten, og reduksjonsbeløpet i gjeld som skal legges inn som nettoformue."
        />

        <br/>

        <Input
          title="Fradrag (ikke minstefradrag)"
          type="text"
          name="fradrag"
          value={stateFradrag || ''}
          onChange={(e)=>this.setState({ stateFradrag:e.target.value})}
          text="Standardfradragene beregnes automatisk. Men har du egne fradrag i tillegg, føres de opp her. De vanligste fradragene er renteutgifter til lån, foreldrefradrag (typisk barnehage og SFO inntil 25.000 kroner for første barn og 15.000 for påfølgende) og pendlerfradrag. Også tap ved salg av aksjer og aksjefond regnes med her."
        />
        <hr/>
        <TaxCalculations {...this.state} />

        <Input
          title="DEL"
          type="text"
          name="del"
          onChange={() => null}
          value={`${window.location.origin}${shareUrl}`}
        />
      </div>
    );
  }
}

Skatt.propTypes = {
  stateFinnmark: PropTypes.bool,
  stateIncome: PropTypes.number,
  stateFormue: PropTypes.number,
  stateFradrag: PropTypes.number,
  stateMarried: PropTypes.bool,
  stateKapital: PropTypes.number,
  taxToPay: PropTypes.object
};

export default Skatt;

