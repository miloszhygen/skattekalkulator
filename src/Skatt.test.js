// Import dependencies
import React from 'react';
import {shallow} from 'enzyme';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

// Import components
import Skatt from './Skatt';

describe('Skatt page', () => {
  const skatt = shallow(
    <Skatt/>
  );
  it('renders properly', () => {
    expect(skatt).toMatchSnapshot();
  });
});
