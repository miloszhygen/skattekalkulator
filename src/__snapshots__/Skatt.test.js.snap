// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skatt page renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Skatt />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Helpers />,
        <header>
          <h1>
            Skattekalkulator 2018
          </h1>
          <h3>
            Skattekalkulator for allminnelig lønnsintekt*
          </h3>
        </header>,
        <Input
          name="income"
          onChange={[Function]}
          title="Skriv inn din bruttoinntekt 2018:"
          type="text"
          value=""
        />,
        <Input
          checked={true}
          onChange={[Function]}
          titleright="Årlig"
          type="radio"
          value="YEARLY"
        />,
        <Input
          checked={false}
          onChange={[Function]}
          titleright="Månedlig"
          type="radio"
          value="MONTHLY"
        />,
        <br />,
        false,
        <hr />,
        <Input
          checked={false}
          onChange={[Function]}
          titleright="Rett til Finnmarksfradrags"
          type="checkbox"
          value="1000"
        />,
        <Input
          checked={false}
          onChange={[Function]}
          titleright="Gift"
          type="checkbox"
          value="1000"
        />,
        <br />,
        <Input
          name="formue"
          onChange={[Function]}
          text="Rente- og kapitalinntekter er typisk renten på bankinnskudd, avkastning på pengemarkedsfond og obligasjonsfond, samt gevinst ved salg av aksjer og aksjefond."
          title="Rente og kapitalinntekter"
          type="text"
          value=""
        />,
        <br />,
        <Input
          name="formue"
          onChange={[Function]}
          text="Det er netto formue som skal legges inn. Fra og med skatteåret 2017 er det visse formuesobjekter som får en verdsettingsrabatt, samt at disse får tilordnet gjeld. Det er formuen etter fradraget for verdsettingsrabatten, og reduksjonsbeløpet i gjeld som skal legges inn som nettoformue."
          title="Din netto formue"
          type="text"
          value=""
        />,
        <br />,
        <Input
          name="fradrag"
          onChange={[Function]}
          text="Standardfradragene beregnes automatisk. Men har du egne fradrag i tillegg, føres de opp her. De vanligste fradragene er renteutgifter til lån, foreldrefradrag (typisk barnehage og SFO inntil 25.000 kroner for første barn og 15.000 for påfølgende) og pendlerfradrag. Også tap ved salg av aksjer og aksjefond regnes med her."
          title="Fradrag (ikke minstefradrag "
          type="text"
          value=""
        />,
        <hr />,
        <TaxCalculations
          incomePr="YEARLY"
          stateFinnmark={false}
          stateFormue={0}
          stateFradrag={0}
          stateIncome={0}
          stateKapital={0}
          stateMarried={false}
          taxToPay={
            Object {
              "formueTotal": 0,
              "income": 0,
              "incomeMonthlyNetto": 0,
              "kommune": 0,
              "minstefradrag": 31800,
              "socialSecurityTax": 0,
              "stat": 0,
              "tax": 0,
              "taxBase": -86550,
              "taxBeforeTrinn": 0,
              "trinnTax": Object {
                "totalTrinnSkatt": 0,
                "trinnFour": 0,
                "trinnOne": 0,
                "trinnThree": 0,
                "trinnTwo": 0,
              },
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Skattekalkulator 2018
            </h1>,
            <h3>
              Skattekalkulator for allminnelig lønnsintekt*
            </h3>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Skattekalkulator 2018",
            },
            "ref": null,
            "rendered": "Skattekalkulator 2018",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Skattekalkulator for allminnelig lønnsintekt*",
            },
            "ref": null,
            "rendered": "Skattekalkulator for allminnelig lønnsintekt*",
            "type": "h3",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "income",
          "onChange": [Function],
          "title": "Skriv inn din bruttoinntekt 2018:",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": true,
          "onChange": [Function],
          "titleright": "Årlig",
          "type": "radio",
          "value": "YEARLY",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": false,
          "onChange": [Function],
          "titleright": "Månedlig",
          "type": "radio",
          "value": "MONTHLY",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": false,
          "onChange": [Function],
          "titleright": "Rett til Finnmarksfradrags",
          "type": "checkbox",
          "value": "1000",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": false,
          "onChange": [Function],
          "titleright": "Gift",
          "type": "checkbox",
          "value": "1000",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "formue",
          "onChange": [Function],
          "text": "Rente- og kapitalinntekter er typisk renten på bankinnskudd, avkastning på pengemarkedsfond og obligasjonsfond, samt gevinst ved salg av aksjer og aksjefond.",
          "title": "Rente og kapitalinntekter",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "formue",
          "onChange": [Function],
          "text": "Det er netto formue som skal legges inn. Fra og med skatteåret 2017 er det visse formuesobjekter som får en verdsettingsrabatt, samt at disse får tilordnet gjeld. Det er formuen etter fradraget for verdsettingsrabatten, og reduksjonsbeløpet i gjeld som skal legges inn som nettoformue.",
          "title": "Din netto formue",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "fradrag",
          "onChange": [Function],
          "text": "Standardfradragene beregnes automatisk. Men har du egne fradrag i tillegg, føres de opp her. De vanligste fradragene er renteutgifter til lån, foreldrefradrag (typisk barnehage og SFO inntil 25.000 kroner for første barn og 15.000 for påfølgende) og pendlerfradrag. Også tap ved salg av aksjer og aksjefond regnes med her.",
          "title": "Fradrag (ikke minstefradrag ",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "incomePr": "YEARLY",
          "stateFinnmark": false,
          "stateFormue": 0,
          "stateFradrag": 0,
          "stateIncome": 0,
          "stateKapital": 0,
          "stateMarried": false,
          "taxToPay": Object {
            "formueTotal": 0,
            "income": 0,
            "incomeMonthlyNetto": 0,
            "kommune": 0,
            "minstefradrag": 31800,
            "socialSecurityTax": 0,
            "stat": 0,
            "tax": 0,
            "taxBase": -86550,
            "taxBeforeTrinn": 0,
            "trinnTax": Object {
              "totalTrinnSkatt": 0,
              "trinnFour": 0,
              "trinnOne": 0,
              "trinnThree": 0,
              "trinnTwo": 0,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Helpers />,
          <header>
            <h1>
              Skattekalkulator 2018
            </h1>
            <h3>
              Skattekalkulator for allminnelig lønnsintekt*
            </h3>
          </header>,
          <Input
            name="income"
            onChange={[Function]}
            title="Skriv inn din bruttoinntekt 2018:"
            type="text"
            value=""
          />,
          <Input
            checked={true}
            onChange={[Function]}
            titleright="Årlig"
            type="radio"
            value="YEARLY"
          />,
          <Input
            checked={false}
            onChange={[Function]}
            titleright="Månedlig"
            type="radio"
            value="MONTHLY"
          />,
          <br />,
          false,
          <hr />,
          <Input
            checked={false}
            onChange={[Function]}
            titleright="Rett til Finnmarksfradrags"
            type="checkbox"
            value="1000"
          />,
          <Input
            checked={false}
            onChange={[Function]}
            titleright="Gift"
            type="checkbox"
            value="1000"
          />,
          <br />,
          <Input
            name="formue"
            onChange={[Function]}
            text="Rente- og kapitalinntekter er typisk renten på bankinnskudd, avkastning på pengemarkedsfond og obligasjonsfond, samt gevinst ved salg av aksjer og aksjefond."
            title="Rente og kapitalinntekter"
            type="text"
            value=""
          />,
          <br />,
          <Input
            name="formue"
            onChange={[Function]}
            text="Det er netto formue som skal legges inn. Fra og med skatteåret 2017 er det visse formuesobjekter som får en verdsettingsrabatt, samt at disse får tilordnet gjeld. Det er formuen etter fradraget for verdsettingsrabatten, og reduksjonsbeløpet i gjeld som skal legges inn som nettoformue."
            title="Din netto formue"
            type="text"
            value=""
          />,
          <br />,
          <Input
            name="fradrag"
            onChange={[Function]}
            text="Standardfradragene beregnes automatisk. Men har du egne fradrag i tillegg, føres de opp her. De vanligste fradragene er renteutgifter til lån, foreldrefradrag (typisk barnehage og SFO inntil 25.000 kroner for første barn og 15.000 for påfølgende) og pendlerfradrag. Også tap ved salg av aksjer og aksjefond regnes med her."
            title="Fradrag (ikke minstefradrag "
            type="text"
            value=""
          />,
          <hr />,
          <TaxCalculations
            incomePr="YEARLY"
            stateFinnmark={false}
            stateFormue={0}
            stateFradrag={0}
            stateIncome={0}
            stateKapital={0}
            stateMarried={false}
            taxToPay={
              Object {
                "formueTotal": 0,
                "income": 0,
                "incomeMonthlyNetto": 0,
                "kommune": 0,
                "minstefradrag": 31800,
                "socialSecurityTax": 0,
                "stat": 0,
                "tax": 0,
                "taxBase": -86550,
                "taxBeforeTrinn": 0,
                "trinnTax": Object {
                  "totalTrinnSkatt": 0,
                  "trinnFour": 0,
                  "trinnOne": 0,
                  "trinnThree": 0,
                  "trinnTwo": 0,
                },
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Skattekalkulator 2018
              </h1>,
              <h3>
                Skattekalkulator for allminnelig lønnsintekt*
              </h3>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Skattekalkulator 2018",
              },
              "ref": null,
              "rendered": "Skattekalkulator 2018",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Skattekalkulator for allminnelig lønnsintekt*",
              },
              "ref": null,
              "rendered": "Skattekalkulator for allminnelig lønnsintekt*",
              "type": "h3",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "income",
            "onChange": [Function],
            "title": "Skriv inn din bruttoinntekt 2018:",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": true,
            "onChange": [Function],
            "titleright": "Årlig",
            "type": "radio",
            "value": "YEARLY",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": false,
            "onChange": [Function],
            "titleright": "Månedlig",
            "type": "radio",
            "value": "MONTHLY",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": false,
            "onChange": [Function],
            "titleright": "Rett til Finnmarksfradrags",
            "type": "checkbox",
            "value": "1000",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": false,
            "onChange": [Function],
            "titleright": "Gift",
            "type": "checkbox",
            "value": "1000",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "formue",
            "onChange": [Function],
            "text": "Rente- og kapitalinntekter er typisk renten på bankinnskudd, avkastning på pengemarkedsfond og obligasjonsfond, samt gevinst ved salg av aksjer og aksjefond.",
            "title": "Rente og kapitalinntekter",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "formue",
            "onChange": [Function],
            "text": "Det er netto formue som skal legges inn. Fra og med skatteåret 2017 er det visse formuesobjekter som får en verdsettingsrabatt, samt at disse får tilordnet gjeld. Det er formuen etter fradraget for verdsettingsrabatten, og reduksjonsbeløpet i gjeld som skal legges inn som nettoformue.",
            "title": "Din netto formue",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "fradrag",
            "onChange": [Function],
            "text": "Standardfradragene beregnes automatisk. Men har du egne fradrag i tillegg, føres de opp her. De vanligste fradragene er renteutgifter til lån, foreldrefradrag (typisk barnehage og SFO inntil 25.000 kroner for første barn og 15.000 for påfølgende) og pendlerfradrag. Også tap ved salg av aksjer og aksjefond regnes med her.",
            "title": "Fradrag (ikke minstefradrag ",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "incomePr": "YEARLY",
            "stateFinnmark": false,
            "stateFormue": 0,
            "stateFradrag": 0,
            "stateIncome": 0,
            "stateKapital": 0,
            "stateMarried": false,
            "taxToPay": Object {
              "formueTotal": 0,
              "income": 0,
              "incomeMonthlyNetto": 0,
              "kommune": 0,
              "minstefradrag": 31800,
              "socialSecurityTax": 0,
              "stat": 0,
              "tax": 0,
              "taxBase": -86550,
              "taxBeforeTrinn": 0,
              "trinnTax": Object {
                "totalTrinnSkatt": 0,
                "trinnFour": 0,
                "trinnOne": 0,
                "trinnThree": 0,
                "trinnTwo": 0,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
